lib/portage/checksum.py

```diff
@@ -458,11 +456,11 @@ def _apply_hash_filter(digests, hash_filter):
                     break
 
     if modified:
-        digests = dict(
-            (k, v)
+        digests = {
+            k: v
             for (k, v) in digests.items()
             if k == "size" or k in verifiable_hash_types
```

Comment:
I wonder if we could optimize this by adding `size` back to `verifiable_hash_types`, or storing the original set (frozen?) earlier on.

Reply:
We could make another variable with the value of `verifiable_hash_types.symmetric_difference({"size"})` which will be the set without `"size"` item. I don't know if that would be a good option, since it could return a deep copy of a new set though.

Reply:
Resolved? 😜

Reply:
Yeah, don't worry about my comments.

Reply:
That's not what I meant. I think I managed to find a solution to your request for that optimization and I wanted your feedback on the changes. What I did was move the `verifiable_hash_types.discard("size")` expression in the conditional that follows. I realized that might have been the primary point of removing it for. After that I was able take out the `k == "size"` boolean expression. I took out the need for using `list()` for the iterator inside that same conditional as well, since a set is already an iterable. Also changed the original set to a frozenset.